frontend/src/components/TaskCard.tsx

```diff
@@ -0,0 +1,172 @@
+import * as React from 'react'
+import styled from 'styled-components'
+import Card from './Card'
+import { Header, Subheader, Paragraph } from './Typography'
+import Button from './Button'
+
+const ExpandedTaskCardWrapper = styled.div`
+    padding: 0px 1.25em 1.25em 1.25em
+    align-content: space-evenly;
+    grid-column: 1 / span 4;
+    grid-row: span 4
+`
+
+const ButtonWrapper = styled.div`
+  grid-column: span 3 / 2;
+  display: flex
+  justify-content: right;
+`
+
+const Input = styled.input`
+  border: none;
+  display: block;
+  font-family: inherit;
+  background-color: inherit;
+  font-size: inherit;
+  width: 100%;
+  padding: 0.5em;
+  margin-bottom: 1.5em;
+  margin-top: -0.5em;
+`
+
+const CompactTaskCardWrapper = styled.div`
+  padding: 0px 20px 20px 20px
+  align-content: space-evenly;
+  grid-column: 1 / span 4;
+`
+
+export interface TaskCardState {
+  expanded: boolean
+  editedTask: {
+    name: string
+    dueDate: string
+    isActive: boolean
+    isRecurrent: boolean
+    flags: string
+    description: string
+  }
+  uneditedTask: {
```

Comment:
Having an `uneditedTask` state object introduces duplication of code (both `editedTask` and `uneditedTask` are initialized with the same values but nothing is ever done to `uneditedTask`). Notice that `uneditedTask` is simply the component props itself. So we can actually remove `uneditedTask` and upon cancelling any changes made, we just set `editedTask` back to props